src/Query/ResultPrinters/TableResultPrinter.php

```diff
@@ -347,6 +350,34 @@ protected function getCellContent( array $dataValues, $outputMode, $isSubject )
 		return $html;
 	}
 
+	/**
+	 * @param @param SMWDataValue $value
+	 * @param int $outputMode
+	 * @param Linker|null|bool $linker
+	 * @return string
+	 */
+	protected function getDataValueText( $value, $outputMode, $isSubject ) {	
+		$prefix = $this->params['prefix'];
+		$linker = $this->getLinker( $isSubject );
+
+		if ( !$prefix || $prefix === 'none' ) {
+			$text = $value->getShortText( $outputMode, $linker );
+
+		} elseif ( $prefix === 'all' || ( $prefix === 'subject' && $isSubject ) ) {
+			$text = $value->getLongText( $outputMode, $linker );
+
+		} elseif ( $prefix === 'auto' && $isSubject ) {
+			$text = $this->mixedResults ? $value->getLongText( $outputMode, $linker ) : 
+				$value->getShortText( $outputMode, $linker );
+
+		} else {
+			$text = $value->getShortText( $outputMode, $linker );
+		}
+
+		return $text;
```

Comment:
This looks very similar to what you added to ValueTextsBuilder, and like the type of duplication that is actually bad by leading to inconsistencies and shotgun surgery.

Reply:
fixed using the trait